src/components/calcite-date/calcite-date.tsx

```diff
@@ -127,68 +183,83 @@ export class CalciteDate {
-              onCalciteInputInput={(e) => this.input(e.detail.value)}
-              placeholder={this.localeData.placeholder}
-              scale={this.scale}
-              type="text"
-              value={formattedDate}
-            />
-          </div>
-        )}
         {this.localeData && (
-          <div class="calendar-picker-wrapper">
-            <calcite-date-month-header
-              activeDate={activeDate}
-              dir={dir}
-              intlNextMonth={this.intlNextMonth}
-              intlPrevMonth={this.intlPrevMonth}
-              localeData={this.localeData}
-              max={max}
-              min={min}
-              onCalciteActiveDateChange={(e: CustomEvent<Date>) => {
-                this.setValue(new Date(e.detail));
-                this.activeDate = new Date(e.detail);
-                this.calciteDateChange.emit(new Date(e.detail));
-              }}
-              scale={this.scale}
-              selectedDate={date || new Date()}
-            />
-            <calcite-date-month
-              activeDate={activeDate}
-              dir={dir}
-              localeData={this.localeData}
-              max={max}
-              min={min}
-              onCalciteActiveDateChange={(e: CustomEvent<Date>) => {
-                this.activeDate = new Date(e.detail);
-              }}
-              onCalciteDateSelect={(e: CustomEvent<Date>) => {
-                this.setValue(new Date(e.detail));
-                this.activeDate = new Date(e.detail);
-                this.calciteDateChange.emit(new Date(e.detail));
-                this.reset();
-              }}
-              scale={this.scale}
-              selectedDate={date}
-            />
+          <div class="input-container" role="application">
+            <div class="input-wrapper">
+              {(this.range || !this.noCalendarInput) && (
+                <calcite-input
+                  class="input"
+                  icon="calendar"
+                  number-button-type="none"
+                  onCalciteInputBlur={(e) => this.blur(e.detail)}
+                  onCalciteInputFocus={() => {
+                    this.active = true;
+                    this.focusedInput = "start";
+                  }}
+                  onCalciteInputInput={(e) => this.input(e.detail.value)}
+                  placeholder={this.localeData?.placeholder}
+                  scale={this.scale}
+                  type="text"
+                  value={formattedDate}
+                />
+              )}
+              {(!this.range || (this.focusedInput === "start" && this.layout !== "vertical")) &&
+                this.renderCalendar(activeDate, dir, maxDate, minDate, date, endDate)}
+            </div>
+            {this.range && this.layout === "horizontal" && (
+              <div class="horizontal-arrow-container">
+                <calcite-icon icon={dir === "ltr" ? `arrow-right` : `arrow-left`} scale="s" />
```

Comment:
you can also do `icon="arrow-right" mirrored` (although that prop is about to change to be called `flip-rtl`.

Reply:
oh that's cool.  Didn't know that.  If I use it, when the prop name change will it automatically refactor my usage too or should I wait to use it after the change.

Reply:
Just saw @macandcheese 's comment that the prop name changed was merged, so I'll update this to use `flip-rtl` .